### PR TITLE
Create circle_sorting.cpp

### PR DESCRIPTION
**Pull request by akshaysharma2277**
**Changes:**
New algo of circle sort.

### CODE REVIEW

File: code/sorting/circle_sort/circle_sorting.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+#include<bits/stdc++.h>
```
Comment (arnavb) [inline code/sorting/circle_sort/circle_sorting.cpp]: Don't include this. See #1915.

File: code/sorting/circle_sort/circle_sorting.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+using namespace std;
+
+// Function to perform circular swaps recursively
+// This function returns true if there was a swap
+// operation performed.
+bool circleSortRec(int a[], int low, int high)
```
Comment (arnavb) [inline code/sorting/circle_sort/circle_sorting.cpp]: vectors are better than arrays.

File: code/sorting/circle_sort/circle_sorting.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    return swapped || firstHalf || secondHalf;
+}
+
+// This function mainly calls circleSortRec
+void circleSort(int a[], int n)
+{
+    // Keep calling circleSortRec while
+    // there is a swap operation.
+    while (circleSortRec(a, 0, n-1))
```
Comment (arnavb) [inline code/sorting/circle_sort/circle_sorting.cpp]: simplify to one line: `while (circleSortRec(a, 0, n-1));`

**Comment by arnavb** [comment]
@AdiChat this implementation already exists; this can be closed.

**Comment by AdiChat** [comment]
@akshaysharma2277 We have closed this pull request but you can open a new pull request for any change anytime. :smile:

**Pull request closed without merging**